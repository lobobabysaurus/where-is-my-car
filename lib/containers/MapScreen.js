import { Location } from 'expo'
import React from 'react'
import { Alert, View } from 'react-native'

import { getCoords, saveCoords } from '../coords'
import Header, { HeaderButton } from '../components/Header'
import MarkButton from '../components/MarkButton'
import MarkedMap from '../components/MarkedMap'
import * as routes from '../routes'
import styles from '../styles'

export default class MapScreen extends React.Component {

  static navigationOptions = ({navigation}) => {
    const historyButton =
      <HeaderButton onPress={() => navigation.navigate(routes.HISTORY)}>
          History
      </HeaderButton>
    return {
      header: <Header title="Map" rightContent={historyButton}/>
    }
  }

  region = {
    latitude: 39.973960,
    latitudeDelta: 0.0111,
    longitude: -75.130620,
    longitudeDelta: 0.0111,
  }

  state = {
    allCoords: [],
    markedCoords: undefined,
  }

  componentDidMount = async () => {
    const coords = await getCoords()
    if (coords) {
      this._updateCoordinates(coords)
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <MarkedMap coords={this.state.markedCoords} region={this.region}/>
        <MarkButton onPress={this._trackLocation} />
      </View>
    )
  }

  _trackLocation = async () => {
    try {
      const position = await Location.getCurrentPositionAsync()
      const coords = await saveCoords(position.coords)
      this._updateCoordinates(coords)
    } catch (err) {
      Alert.alert(`Error getting location: ${err}`)
    }
  }

  _updateCoordinates = (coords) => {
    this.setState({
      allCoords: coords,
      markedCoords: coords[coords.length-1]
    })
  }
}
