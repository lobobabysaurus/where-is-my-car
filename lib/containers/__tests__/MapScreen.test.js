import React from 'react'
import renderer from 'react-test-renderer'

import MapScreen from '../MapScreen'

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react')
  const MapView = require.requireActual('react-native-maps')

  const MockCallout = (props) => (
    React.createElement('Callout', props, props.children)
  )

  const MockMarker = (props) => (
    React.createElement('Marker', props, props.children)
  )

  const MockMapView = (props) => (
    React.createElement('MapView', props, props.children)
  )

  MockCallout.propTypes = MapView.Callout.propTypes
  MockMarker.propTypes = MapView.Marker.propTypes
  MockMapView.propTypes = MapView.propTypes
  MockMapView.Marker = MockMarker
  MockMapView.Callout = MockCallout
  return MockMapView
})


describe('MapScreen', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<MapScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
