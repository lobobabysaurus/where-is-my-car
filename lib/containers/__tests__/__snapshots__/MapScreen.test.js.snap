// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <MapView
    initialRegion={
      Object {
        "latitude": 39.97396,
        "latitudeDelta": 0.0111,
        "longitude": -75.13062,
        "longitudeDelta": 0.0111,
      }
    }
    style={
      Object {
        "flexBasis": "90%",
      }
    }
  />
  <ExponentLinearGradient
    colors={
      Array [
        4283484280,
        4278190219,
      ]
    }
    style={
      Object {
        "flexBasis": "10%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 36,
          }
        }
      >
        Mark
      </Text>
    </View>
  </ExponentLinearGradient>
</View>
`;
